keras/regularizers.py

```diff
@@ -1,26 +1,80 @@
 from __future__ import absolute_import
-import theano
 import theano.tensor as T
-import numpy as np
-
-def l1(l=.01):
-    def l1wrap(g, p):
-        g += T.sgn(p) * l
-        return g
-    return l1wrap
-
-def l2(l=.01):
-    def l2wrap(g, p):
-        g += p * l
-        return g
-    return l2wrap
-
-def l1l2(l1=.01, l2=.01):
-    def l1l2wrap(g, p):
-        g += T.sgn(p) * l1
-        g += p * l2
-        return g
-    return l1l2wrap
-
-def identity(g, p):
-    return g
\ No newline at end of file
+
+
+class Regularizer(object):
+
+    def update_gradient(self, gradient, params):
+        raise NotImplementedError
+
+    def update_loss(self, loss):
+        raise NotImplementedError
+
+
+class WeightsL1(Regularizer):
+    def __init__(self, l=0.01):
+        self.l = l
+
+    def update_gradient(self, gradient, params):
+        gradient += T.sgn(params) * self.l
+        return gradient
+
+
+class WeightsL2(Regularizer):
+    def __init__(self, l=0.01):
+        self.l = l
+
+    def update_gradient(self, gradient, params):
+        gradient += params * self.l
+        return gradient
+
+
+class WeightsL1L2(Regularizer):
+    def __init__(self, l1=0.01, l2=0.01):
+        self.l1 = l1
+        self.l2 = l2
+
+    def update_gradient(self, gradient, params):
+        gradient += params * self.l2
+        gradient += T.sgn(params) * self.l1
+        return gradient
+
+
+class Identity(Regularizer):
```

Comment:
Indeed, actually could do that in the base class.

But do we not maybe want to keep it like this so that an error gets raised in the event that one of these regularizers gets put into a place where it does not belong? (like putting an activity regularizer in the slot for a weight regularizer?)